avocado/utils/archive.py

```diff
@@ -157,6 +163,27 @@ def extract(self, path='.'):
         :param path: destination path.
         """
         self._engine.extractall(path)
+        if self.is_zip:
+            self._update_symlinks(path)
+
+    def _update_symlinks(self, dst_dir='.'):
+        if platform.system() != "Linux":
+            LOG.warn("Symlink handling in zip files not supported on Windows.")
+            return
+        # Walk all files and re-create files as symlinks
+        for path, info in self._engine.NameToInfo.iteritems():
+            if info.external_attr & 2684354560 == 2684354560:
```

Comment:
Well I considered using `unzip` utility instead, which would be probably simpler, but I'm not sure we want to do depend on yet another utility.

Regarding upstream there is only the permission issue from 2012 http://bugs.python.org/issue15795 so I submitted a new one for symlinks http://bugs.python.org/issue27318 to find out how they react. If that goes well I'll try to send the perm fix to the other issue as well.